Source/OxyPlot/Series/XYAxisSeries.cs

```diff
@@ -750,62 +750,65 @@ protected int UpdateWindowStartIndex<T>(IList<T> items, Func<T, double> xgetter,
         /// <param name="targetX">target x.</param>
         /// <param name="initialGuess">initial guess index.</param>
         /// <returns>
-        /// index of x with max(x) &lt;= target x or -1 if cannot find
+        /// index of x with max(x) &lt;= target x or 0 if cannot find
         /// </returns>
-        protected int FindWindowStartIndex<T>(IList<T> items, Func<T, double> xgetter, double targetX, int initialGuess)
+        public int FindWindowStartIndex<T>(IList<T> items, Func<T, double> xgetter, double targetX, int initialGuess)
         {
-            int lastguess = 0;
             int start = 0;
-            int end = items.Count - 1;
-            int curGuess = initialGuess;
+            int nominalEnd = items.Count - 1;
+            while (nominalEnd > 0 && double.IsNaN(xgetter(items[nominalEnd])))
+                nominalEnd -= 1;
+            int end = nominalEnd;
+            int curGuess = Math.Max(0, Math.Min(end, initialGuess));
 
-            while (start <= end)
+            double GetX(int index)
             {
-                if (curGuess < start)
+                while (index <= nominalEnd)
                 {
-                    return lastguess;
-                }
-                else if (curGuess > end)
-                {
-                    return end;
+                    double guessX = xgetter(items[index]);
+                    if (double.IsNaN(guessX))
+                        index += 1;
+                    else
+                        return guessX;
                 }
+                return xgetter(items[nominalEnd]);
+            }
 
-                double guessX = xgetter(items[curGuess]);
+            while (start < end)
+            {
+                double guessX = GetX(curGuess);
                 if (guessX.Equals(targetX))
```

Comment:
(Not an issue with this PR, but may be relevant to the issue). This logic is inconsistent with the assumption of monotonicity: it is possible for a previous `X` value to also equal `targetX`.

Reply:
Yeah, I think that equals-comparison is quite suspect. My thought was that it might be some kind of optimization - perhaps we end up there if the function is called with a 'correct' initial guess?

I think the idea of monotonicity in oxyplot is a graph where the Y-value is a function of the X-value?

In that case, what we really have is strict monotonicity - it shouldn't be allowed to have two different values with the same X-value, since the curve wouldn't correspond to a function of the X-axis in that case.

Then of course we have parametric graphs where both X and Y depend on an independent variable, in that case the X-axis values wouldn't necessarily have any monotonicity.

My proposal would be to change the interpretation of Monotonicity in Oxyplot and replace it with strict monotonicity.